app/models/king.rb

```diff
@@ -1,12 +1,44 @@
 class King < Piece
 
+	def can_castle?(x_destination, y_destination)
+		# Castling allowed if:
+		# king and rook not moved
+		# no pieces between king and rook
+		# king not in check either at start or at destination
+		@game = self.game
+		@has_moved != true && !obstructed?([x_destination, y_destination]) && @game.in_check?(color) == false
```

Comment:
If you are changing the `in_check?(color)` method to `in_check?(current_color)` then this line should be updated because it's still using the old method.